File: VB/App_Code/ThemeModel.vb
Option Infer On

Imports System
Imports System.Xml.Serialization
Imports System.Collections.Generic
Imports System.Web.UI
Imports System.Drawing



Public Class ThemeModel
    Inherits ThemeModelBase

    Private _baseColor As String
    Private _fontFamily As String
    Private _fontSize As String
    Private _showAsTop As Boolean
    Private _previewColor As String
    <XmlAttribute> _
    Public Property BaseColor() As String
        Get
            If String.IsNullOrEmpty(_baseColor) Then
                Return ""
            End If
            Return _baseColor
        End Get
        Set(ByVal value As String)
            _baseColor = value.ToUpper()
        End Set
    End Property
    <XmlAttribute> _
    Public Property IsNew() As Boolean
    <XmlAttribute> _
    Public Property FontFamily() As String
        Get
            If String.IsNullOrEmpty(_fontFamily) Then
                Return ""
            End If
            Return _fontFamily
        End Get
        Set(ByVal value As String)
            _fontFamily = value
        End Set
    End Property
    <XmlAttribute> _
    Public Property FontSize() As String
        Get
            If String.IsNullOrEmpty(_fontSize) Then
                Return ""
            End If
            Return _fontSize
        End Get
        Set(ByVal value As String)
            _fontSize = value
        End Set
    End Property
    Public ReadOnly Property Font() As String
        Get
            Dim result = String.Empty
            If Not String.IsNullOrEmpty(FontSize) AndAlso Not String.IsNullOrEmpty(FontFamily) Then
                result = String.Format("{0} {1}", FontSize, FontFamily)
            End If
            Return result
        End Get
    End Property
    <XmlAttribute> _
    Public Property ShowAsTop() As Boolean
        Get
            Return _showAsTop
        End Get
        Set(ByVal value As Boolean)
            _showAsTop = value
        End Set
    End Property
    <XmlAttribute> _
    Public Property PreveiwColor() As String
        Get
            If String.IsNullOrEmpty(_previewColor) Then
                Return ""
            End If
            Return _previewColor
        End Get
        Set(ByVal value As String)
            _previewColor = value.ToUpper()
        End Set
    End Property
End Class

File: CS/App_Code/ThemeModel.cs
using System;
using System.Xml.Serialization;
using System.Collections.Generic;
using System.Web.UI;
using System.Drawing;



public class ThemeModel : ThemeModelBase {
    string _baseColor;
    string _fontFamily;
    string _fontSize;
    bool _showAsTop;
    string _previewColor;
    [XmlAttribute]
    public string BaseColor {
        get {
            if(String.IsNullOrEmpty(_baseColor))
                return "";
            return _baseColor;
        }
        set {
            _baseColor = value.ToUpper();
        }
    }
    [XmlAttribute]
    public bool IsNew { get; set; }
    [XmlAttribute]
    public string FontFamily {
        get {
            if(String.IsNullOrEmpty(_fontFamily))
                return "";
            return _fontFamily;
        }
        set {
            _fontFamily = value;
        }
    }
    [XmlAttribute]
    public string FontSize {
        get {
            if(String.IsNullOrEmpty(_fontSize))
                return "";
            return _fontSize;
        }
        set {
            _fontSize = value;
        }
    }
    public string Font {
        get {
            var result = string.Empty;
            if(!string.IsNullOrEmpty(FontSize) && !string.IsNullOrEmpty(FontFamily))
                result = string.Format("{0} {1}", FontSize, FontFamily);
            return result;
        }
    }
    [XmlAttribute]
    public bool ShowAsTop {
        get {
            return _showAsTop;
        }
        set {
            _showAsTop = value;
        }
    }
    [XmlAttribute]
    public string PreveiwColor {
        get {
            if(String.IsNullOrEmpty(_previewColor))
                return "";
            return _previewColor;
        }
        set {
            _previewColor = value.ToUpper();
        }
    }
}


